### ISSUE TITLE
[Accessibility] Aria-Selected Attribute missing from Accordion

### ISSUE THREAD

**Issue by anpan01** [zurb/foundation-sites#11437]
## What should happen?
Using accordion should add aria-selected attribute in order to pass AA and AAA accessibility conformance.

## What happens instead?
Accordion is created without aria-selected attribute which causes it to get flagged in SiteImprove Accessibility Checker. 

## Possible Solution
When using accordion class names, add aria-selected="false" attribute to nested <li> elements. 

## Test Case and/or Steps to Reproduce
Test Case:
Here is a screen shot depicting the accessibility issue for the missing aria-selected attribute. 

https://imgur.com/a/THYiONr

How to reproduce:
1. Create menu using accordion class names.
2. Use SiteImprove to test AA and AAA accessibility.
3. Go through SiteImprove to see the alerts for missing aria states or properties.

## Context
Effected myself and my team when trying to create a fully accessible web-based application. 

## Your Environment
- Foundation version(s) used: V6.4.1
- Browser(s) name and version(s): Chrome
- Operating System and version (desktop or mobile): Microsoft Windows 10 Pro

## Checklist (all required):

- [x] I have read and follow the CONTRIBUTING.md document.
- [x] There are no other issues similar to this one.
- [x] The issue title is descriptive.
- [x] The template is correctly filled.

Thank you all for the great framework! Hope this helps improve accessibility for us all.

**Comment by Owlbertz** [comment]
@anpan01 thanks for reporting this issue. I noticed that your title states "Accordion", when your screenshot includes an "Accordion Menu", thus I updated the issue's title.

According to the [W3C documentation](https://www.w3.org/TR/wai-aria-1.1/#treeitem) elements with the role "treeitem" need the `aria-selected`-attribute to be set.
Since the main idea of the Acordion Menu is navigate-ion and thus a redirect, the website would be re-rendered and the state would be lost.
It must be discussed whether it makes sense to always update the `aria-selected` state nonetheless.

**Comment by ncoden** [comment]
@Owlbertz I read some parts of the W3C documentation you brought and IMHO there should be a `aria-selected` set when any link is focused (including parents, which do not redirect when clicked). Plus, the user may navigate with keyboard and "select" one link or an other without following the `[href]`. What do you think ?

**Comment by Owlbertz** [comment]
Based on the [WC3 documentation on aria-selected](https://www.w3.org/TR/wai-aria-1.1/#aria-selected):
> Single-selection containers where the currently focused item is not selected. The selection normally follows the focus, and is managed by the user agent.

The problem here is, that the Accordion Menu is actually not a single selection container because it has `aria-multiselectable="true"`, which requires `aria-selected` to be set, see [W3C documentation on aria-multiselectable](https://www.w3.org/TR/wai-aria-1.1/#aria-multiselectable):
> Authors SHOULD ensure that selected descendants have the aria-selected attribute set to true, and selectable descendant have the aria-selected attribute set to false.

That being said I believe the actual issue is whether the Accordion Menu needs the `aria-multiselectable` flag set or not. Based on my understanding, it should not have that flag set to `true` as "selectable" addresses the tree's leaves (i.e. the `role="treeitem"`, which means the actual links inside the menu) and you cannot have multiple links selected.
(My guess would be that someone added that flag because he assumed that it deals with multiple accordion panels being opened simultaneously).

**Comment by ncoden** [comment]
According to [this article](https://dommagnifi.co/update/2016/09/23/introduction-to-aria-attributes.html) that use an accordion as example, we should use both `aria-expanded` and `aria-selected` when the user navigate through the menu.

I found very few articles talking about both `aria-selected` and `aria-expanded`, and W3C does not talk much about when a component should be considered in a "selected" state.

So for now I'm going to the "both attributes" approach but I'm not confident at all about it.

**Comment by ncoden** [comment]
@Owlbertz According to [this article](https://dommagnifi.co/update/2016/09/23/introduction-to-aria-attributes.html) that use an accordion as example, we should use both `aria-expanded` and `aria-selected` when the user navigate through the menu.

I found very few articles talking about both `aria-selected` and `aria-expanded`, and W3C does not talk much about when a component should be considered in a "selected" state.

So for now I'm going to the "both attributes" approach but I'm not confident at all about it.

**Comment by Owlbertz** [comment]
I am not sure whether the article fits our use case. First of all, we need to differentiate between an [accordion](https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html) and a [tree](https://www.w3.org/TR/wai-aria-practices-1.1/examples/treeview/treeview-1/treeview-1a.html). 

As you can see in the link above, W3C does not use the `aria-selected` in either case.
This is - of course - related to the fact that `aria-multiselectable` isn't used in their tree example. But as stated above I do not think that for our Accordion Menu we need that attribute at all.

Imo we should not use `aria-selected` but remove `aria-multiselectable` instead.

**Comment by JeremyEnglert** [comment]
I think the _actual_ issue here is that we're using `role="tree" on the accordions and accordion menus. @Owlbertz, correct me if I'm wrong, but that doesn't seem right.

**Comment by efecher** [comment]
same issue in practice I'm seeing in SiteImprove where my employer is using Drilldown menus in an off-canvas container for a mobile nav. The drilldown is given "tree" and "treeitem" roles, which apparently is giving the impression it's data represented as a selectable/expandable tree (like a file explorer with a folder view) instead of a menu with submenus. Shouldn't the role be "menu", not "tree" for something like a drilldown?
